packages/astro/src/content/vite-plugin-content-virtual-mod.ts

```diff
@@ -20,98 +21,177 @@ import {
+	let contentEntryGlobResult: string;
+	let dataEntryGlobResult: string;
+	let renderEntryGlobResult: string;
+	if (IS_DEV || IS_SERVER || !settings.config.experimental.contentCollectionCache) {
+		const contentEntryConfigByExt = getEntryConfigByExtMap(settings.contentEntryTypes);
+		const contentEntryExts = [...contentEntryConfigByExt.keys()];
+		const dataEntryExts = getDataEntryExts(settings);
+		const createGlob = (value: string[], flag: string) => `import.meta.glob(${JSON.stringify(value)}, { query: { ${flag}: true } })`
+		contentEntryGlobResult = createGlob(globWithUnderscoresIgnored(relContentDir, contentEntryExts), CONTENT_FLAG);
+		dataEntryGlobResult = createGlob(globWithUnderscoresIgnored(relContentDir, dataEntryExts), DATA_FLAG);
+		renderEntryGlobResult = createGlob(globWithUnderscoresIgnored(relContentDir, contentEntryExts), CONTENT_RENDER_FLAG);
+	} else {
+		contentEntryGlobResult = getStringifiedCollectionFromLookup('content', relContentDir, lookupMap);
+		dataEntryGlobResult = getStringifiedCollectionFromLookup('data', relContentDir, lookupMap);
+		renderEntryGlobResult = getStringifiedCollectionFromLookup('render', relContentDir, lookupMap);
+	}
 
-	const virtualModContents = fsMod
+	const virtualModContents = nodeFs
 		.readFileSync(contentPaths.virtualModTemplate, 'utf-8')
-		.replace(
-			'@@COLLECTION_NAME_BY_REFERENCE_KEY@@',
-			new URL('reference-map.json', contentPaths.cacheDir).pathname
-		)
 		.replace('@@CONTENT_DIR@@', relContentDir)
 		.replace(
 			"'@@CONTENT_ENTRY_GLOB_PATH@@'",
-			JSON.stringify(globWithUnderscoresIgnored(relContentDir, contentEntryExts))
+			contentEntryGlobResult
 		)
 		.replace(
 			"'@@DATA_ENTRY_GLOB_PATH@@'",
-			JSON.stringify(globWithUnderscoresIgnored(relContentDir, dataEntryExts))
+			dataEntryGlobResult
 		)
 		.replace(
 			"'@@RENDER_ENTRY_GLOB_PATH@@'",
-			JSON.stringify(
-				globWithUnderscoresIgnored(
-					relContentDir,
-					/** Note: data collections excluded */ contentEntryExts
-				)
-			)
+			renderEntryGlobResult
+		).replace(
+			'/* @@LOOKUP_MAP_ASSIGNMENT@@ */',
+			`lookupMap = ${JSON.stringify(lookupMap)};`
 		);
 
-	const astroContentVirtualModuleId = '\0' + VIRTUAL_MODULE_ID;
-
-	return {
-		name: 'astro-content-virtual-mod-plugin',
-		resolveId(id) {
-			if (id === VIRTUAL_MODULE_ID) {
-				return astroContentVirtualModuleId;
-			}
-		},
-		async load(id) {
-			if (id === astroContentVirtualModuleId) {
-				const stringifiedLookupMap = await getStringifiedLookupMap({
-					fs: fsMod,
-					contentPaths,
-					contentEntryConfigByExt,
-					dataEntryExts,
-					root: settings.config.root,
-				});
+	return virtualModContents;
+}
 
-				return {
-					code: virtualModContents.replace(
-						'/* @@LOOKUP_MAP_ASSIGNMENT@@ */',
-						`lookupMap = ${stringifiedLookupMap};`
-					),
-				};
+function getStringifiedCollectionFromLookup(wantedType: 'content' | 'data' | 'render', relContentDir: string, lookupMap: ContentLookupMap) {
```

Comment:
It seems we introduced some new logic: `'content' | 'data' | 'render'`. Maybe it's worth having a comment that explains what we do. 

If you think it's too technical, we could also have a `README.md`